src/Attachment.js

```diff
@@ -0,0 +1,176 @@
+        */
+        this.contentType = null;
+
+        /**
+        @property length
+        @type int
+        */
+        this.length = null;
+
+        /**
+        @property sha2
+        @type String
+        */
+        this.sha2 = null;
+
+        /**
+        @property description
+        @type Object
+        */
+        this.description = null;
+
+        /**
+        @property fileUrl
+        @type String
+        */
+        this.fileUrl = null;
+
+        this.init(cfg);
+    };
+    Attachment.prototype = {
+        /**
+        @property LOG_SRC
+        */
+        LOG_SRC: "Attachment",
+
+        /**
+        @method log
+        */
+        log: TinCan.prototype.log,
+
+        /**
+        @method init
+        @param {Object} [options] Configuration used to initialize
+        */
+        init: function (cfg) {
+            this.log("init");
+            var i,
+                directProps = [
+                    "contentType",
+                    "length",
+                    "sha2",
+                    "usageType",
+                    "display"
+                ]
+            ;
+
+            cfg = cfg || {};
+
+            if (cfg.hasOwnProperty("description") && typeof cfg.description !== "undefined" && cfg.description !== null) {
+                this.description = cfg.description;
+            }
+            if (cfg.hasOwnProperty("fileUrl") && typeof cfg.fileUrl !== "undefined" && cfg.fileUrl !== null) {
+                this.fileUrl = cfg.fileUrl;
+            }
+
+            for (i = 0; i < directProps.length; i += 1) {
+                if (cfg.hasOwnProperty(directProps[i]) && cfg[directProps[i]] !== null) {
+                    this[directProps[i]] = cfg[directProps[i]];
+                }
+            }
+
+            if (cfg.hasOwnProperty("content") && typeof cfg.content !== "undefined" && cfg.content !== null) {
+                this.content = cfg.content;
+                this.length = cfg.content.length;
+                this.sha2 = TinCan.Utils.getSHA256String(cfg.content);
+            }
+        },
+
+        /**
+        @method asVersion
+        @param {String} [version] Version to return (defaults to newest supported)
+        */
+        asVersion: function (version) {
+            this.log("asVersion");
+            var result;
+
+            version = version || TinCan.versions()[0];
+
+            if (version === "0.9" || version === "0.95") {
+                result = null;
+            }
+            else {
+                result = {
+                    contentType: this.contentType,
+                    description: this.description,
+                    display: this.display,
+                    fileUrl: this.fileUrl,
+                    length: this.length,
+                    sha2: this.sha2,
+                    usageType: this.usageType
+                };
+                if (this.hasOwnProperty("content") && typeof this.content !== "undefined" && this.content !== null) {
```

Comment:
`this` will *always* have `content` based on the object's prototype based on comment above, this can be reduced to just check for null.